File: php/hardcoded_secrets/payment_config.php
<?php
/**
 * Shopist - Payment Configuration
 * DEMO FILE: Intentionally vulnerable for Datadog Code Security SAST demo.
 * DO NOT USE IN PRODUCTION.
 */

// VULN 1: Stripe live secret key hardcoded in source (Hardcoded Secret)
$stripeKey = "sk_live_4eC39HqLyjWDarjtT1zdp7dc";

\Stripe\Stripe::setApiKey($stripeKey);

function createPaymentIntent($amount, $currency = 'usd') {
    return \Stripe\PaymentIntent::create([
        'amount'   => $amount,
        'currency' => $currency,
    ]);
}

// VULN 2: AWS access key ID and secret hardcoded in S3 client configuration array (Hardcoded Secret)
function getS3Client() {
    return new Aws\S3\S3Client([
        'version'     => 'latest',
        'region'      => 'us-east-1',
        'credentials' => [
            'key'    => 'AKIAIOSFODNN7EXAMPLE',
            'secret' => 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
        ],
    ]);
}

function uploadReceiptToS3($localPath, $orderId) {
    $s3 = getS3Client();
    $s3->putObject([
        'Bucket' => 'shopist-receipts',
        'Key'    => "receipts/{$orderId}.pdf",
        'Body'   => fopen($localPath, 'r'),
    ]);
}

// VULN 3: Database password hardcoded in mysqli_connect call (Hardcoded Secret)
function getPaymentDbConnection() {
    $conn = mysqli_connect(
        "payments-db.shopist.internal",
        "payments_user",
        "Sh0p1st_DB_P@ssw0rd!",   // hardcoded production password
        "shopist_payments"
    );
    if (!$conn) {
        die("Payment DB connection failed: " . mysqli_connect_error());
    }
    return $conn;
}

File: dotnet/hardcoded_secrets/PaymentConfig.cs
using System.Data.SqlClient;
using Amazon.S3;
using Amazon.Runtime;
using Stripe;

public class PaymentConfig
{
    // VULN 1: Hardcoded Stripe secret key in source code
    public StripeClient CreateStripeClient()
    {
        string stripeSecretKey = "sk_live_4eC39HqLyjWDarjtT1zdp7dc";
        return new StripeClient(stripeSecretKey);
    }

    // VULN 2: Hardcoded AWS AccessKey and SecretKey passed directly to AmazonS3Client
    public AmazonS3Client CreateS3Client()
    {
        string accessKey = "AKIAIOSFODNN7EXAMPLE";
        string secretKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY";
        var credentials = new BasicAWSCredentials(accessKey, secretKey);
        return new AmazonS3Client(credentials, Amazon.RegionEndpoint.USEast1);
    }

    // VULN 3: Hardcoded database password in connection string for payment records
    public SqlConnection CreatePaymentDbConnection()
    {
        string connectionString = "Server=payments-db.shopist.internal;Database=payments;User Id=payments_app;Password=Sup3rS3cr3tP@ssw0rd!;";
        return new SqlConnection(connectionString);
    }
}
